admin/adminActionsRestricted.php

```diff
@@ -1152,6 +1158,42 @@ public function showApiKeys($params)
 		";
 	}
 
+	public function calculateGR($params) {
+		require_once(l_r('ghostratings/calculations.php'));
+		global $DB;
+		$batch_size = (int)$params['batchSize'];
+		if ($batch_size <= 0)
+		{
+			return l_t('Invalid batch size. Please enter a positive integer.');
+		}
+		$data = $DB->sql_tabl("SELECT id, variantID, pressType, potType, turn, gameOver, phaseMinutes, processTime, playerTypes FROM wD_Games WHERE gameOver <> 'No' AND grCalculated <> 1 ORDER BY processTime ASC LIMIT ".$batch_size.";");

```

Comment:
I know that the gameOver column and another column in this table have to be used carefully in queries because they aren't indexed. Lets get together at some point and test this select on the db for speed before pushing this.

Reply:
Sounds good. Like most SQL related things, I defer to you completely.